src/app/services/graph-client/auth/authentication-module.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  AccountInfo,
+  AuthenticationResult, InteractionRequiredAuthError,
+  PopupRequest, PublicClientApplication, SilentRequest
+} from '@azure/msal-browser';
+
+import { geLocale } from '../../../../appLocale';
+import { AUTH_URL, DEFAULT_USER_SCOPES } from '../../graph-constants';
+
+const defaultScopes = DEFAULT_USER_SCOPES.split(' ');
+
+export class AuthenticationModule {
+
+  private msalApplication: PublicClientApplication;
+
+  constructor(msalApplication: PublicClientApplication) {
+    this.msalApplication = msalApplication;
+  }
+
+  public getAccount(): AccountInfo | undefined {
+    if (this.msalApplication) {
+      const allAccounts = this.msalApplication.getAllAccounts();
+      if (allAccounts && allAccounts.length > 0) {
+        return allAccounts[0];
```

Comment:
Is it guaranteed that the first account that is the account[0] will always be the valid or intended account?

Reply:
Thanks for pointing this out. I have seen that the MSAL.js team recommends that we track this user ourselves

Reply:
> Is it guaranteed that the first account that is the account[0] will always be the valid or intended account?

Any idea about this? My concern is about picking up the wrong account. It would be a major issue if that happens.

Reply:
Yeah... 
I can use either:

- msalApplication.getAccountByHomeId
- msalApplication.getAccountByLocalId
- msalApplication.getAccountByUsername


This means I'd have to create a way to track either of the identifiers